src/charm_state.py

```diff
@@ -266,6 +268,35 @@ def userids_to_list(cls, value: str) -> typing.List[str]:
                 raise ValidationError(f"Invalid user ID format: {user_id}", cls)
         return value_list
 
+    @validator("ready_check")
+    @classmethod
+    def to_pebble_check(cls, value: str) -> typing.Dict[str, typing.Union[str, int]]:
+        """Convert the ready_check field to pebble check.
+
+        Args:
+            value: the input value.
+
+        Returns:
+            The pebble check.
+
+        Raises:
+            ValidationError: if ready_check is invalid.
+        """
+        # expected
+        # period;threshold;timeout
+        config_values = value.split(",")
+        if len(config_values) < 3:
+            raise ValidationError(f"Invalid ready_check, less than 3 values: {value}", cls)
```

Comment:
What are the semantics for having more than three?

Reply:
Also fixed. Changed to check for != 3
Thanks @cbartz :)